src/strings/ops.c

```diff
@@ -877,6 +882,42 @@ static MVMint64 string_equal_at_ignore_case(MVMThreadContext *tc, MVMString *Hay
         return H_graphs + H_expansion - H_offset >= n_fc_graphs  ? 1 : 0;
     return 0;
 }
+static void knuth_morris_pratt_process_pattern (MVMThreadContext *tc, MVMString *pat, MVMGrapheme32 next[], MVMStringIndex pat_graphs) {
+    MVMint64 i = 0;
+    MVMint64 j = next[0] = -1;
+    while (i < pat_graphs) {
+        if (j == -1 || MVM_string_get_grapheme_at_nocheck(tc, pat, i) == MVM_string_get_grapheme_at_nocheck(tc, pat, j)) {
```

Comment:
Seems like with so many calls to `MVM_string_get_grapheme_at_nocheck` it might be beneficial to flatten the pattern into a MVMGrapheme32[] and access that instead.